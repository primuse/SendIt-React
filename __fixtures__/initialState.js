export const props = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: false
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: false
  }
};

export const props2 = {
  auth: {
    isLoading: false,
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      registered: '22-03-2019',
      email: 'cim@gmail.com',
      isadmin: true
    }
  },
  parcel: {
    parcels: [
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      },
      {
        id: '1',
        parcelName: 'rice',
        weigth: '30',
        price: '4,000',
        destination: 'Owerri',
        receiver: 'Tiku Okoye',
        senton: '22-03-2019',
        status: 'created'
      }
    ],
    createdParcels: [],
    deliveredParcels: [],
    inTransitParcels: [],
    canceledParcels: [],
    isLoading: false,
    userParcel: {
      id: 2,
      parcelName: 'rice',
      weigth: '30',
      price: '4,000',
      destination: 'Owerri',
      receiver: 'Tiku Okoye',
      senton: '22-03-2019',
      status: 'created'
    }
  },
  users: {
    allUsers: [
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      },
      {
        id: '1',
        firstName: 'Tiku',
        lastName: 'Okoye',
        isadmin: 'false',
        email: 'cim@gmail.com',
        registered: '22-03-2019',
      }
    ],
    show: false
  },
  user: {
    firstName: 'Tiku',
    lastName: 'Okoye',
    registered: '22-03-2019',
    email: 'cim@gmail.com',
    isadmin: true
  }
};

export const props3 = {
  parcel: {
    isLoading: true
  }
};
