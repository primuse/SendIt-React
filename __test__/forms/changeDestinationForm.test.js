import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import DestinationParcelForm, { DestinationModalForm }
  from '../../src/components/forms/changeDestinationForm.jsx';
import { props, props3 } from '../../__fixtures__/initialState';

const onChange = jest.fn();
const save = jest.fn();
const changeParcelDestination = jest.fn();
const mockStore = configureStore([thunk]);
const store = mockStore(props);
let wrapper;

describe('Change Parcel Destination Form', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}>
        <DestinationParcelForm />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Change Parcel Destination Form', () => {
  beforeEach(() => {
    wrapper = mount(<DestinationModalForm {...props}
      onChange={onChange} save={save}
      changeParcelDestination={changeParcelDestination} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should change state', () => {
    const input = wrapper.find('#destination');
    input.instance().value = 'USA';
    input.simulate('change');
    expect(wrapper.state().destination).toEqual('USA');
  });
  it('should simulate the onSubmit', () => {
    wrapper.find('form').simulate('submit');
    const input1 = wrapper.find('input').at(0);
    input1.simulate('change', { target: { value: 'USA' } });
    expect(changeParcelDestination).toHaveBeenCalled();
  });
});

describe('Change Parcel Destination Form', () => {
  beforeEach(() => {
    wrapper = mount(<DestinationModalForm {...props3}
      onChange={onChange} save={save}
      changeParcelDestination={changeParcelDestination} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should show the loading button', () => {
    expect(wrapper.find('span')).toBeDefined();
  });
});
