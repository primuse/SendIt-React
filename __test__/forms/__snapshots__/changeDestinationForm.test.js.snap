// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New parcel form renders correctly 1`] = `
<form
  className="text-right"
  id="changeDestination"
  onSubmit={[Function]}
>
  <div
    className="form-row"
  >
    <input
      id="destination"
      name="destination"
      onChange={[Function]}
      required={true}
      type="text"
      value=""
    />
  </div>
  <button
    className="btn xsm is-outlined white"
    type="submit"
  >
    Save
  </button>
</form>
`;
